uwblueprint/paramedics-react
# Add event dashboard page

## Overview
- Added Event Dashboard Page
- Connected menu to CCP Dashboard

[Notion task](https://www.notion.so/uwblueprintexecs/Event-Dashboard-Screen-af7c8340fc394304abc1cabaa5ae2296)

- TO DO (separately): Add and update `MenuAppBar` to fit in CCP Dashboard (highlight corresponding checkpoint in menu)

## Changes
- Added route for Event Dashboard Page (at `/events/{eventId}`)
-  Add Event Dashboard Page with CCP, Hospital, Ambulance tabs
    - Tables in each tab panel use real data
    - CCP table doesn't actually have addresses
    - Functionality of including/excluding multiple and single hospitals/ambulances
    - Included select all and table sort 
- Display event specific info (no addresses right now)
- Hamburger menu with contents

## Testing
1. Go to an event dashboard page via the route `/events/{eventId}`
2. Check that the event name and date are correct
3. (Add some CCPs, Hospitals, and Ambulances to the event in the backend to verify they show up and have the correct active/inactive status and include/exclude button)
4. On the Hospital and Ambulance tabs, the select all box should select all table rows and the number selected should be updated accordingly. Verify that the include/exclude buttons on each table row are disabled and that include/exclude buttons appear above the table.
5. Try including/excluding hospitals via selecting checkboxes. 
    - Check that the confirmation dialog includes the selected hospitals.
    - Cancel should close dialog and deselect
    - Confirm should include/exclude (status should update without page refresh)
6. Try including/excluding a single hospital using the button in a table row.
7. Try including/excluding ambulances via selecting checkboxes. Check that the confirmation dialog includes the selected hospitals.
8. Try including/excluding a single ambulance using the button in a table row.
9. Verify the sort functionality on the table columns work.

## Screenshots
CCP tab:
![image](https://user-images.githubusercontent.com/22457859/89206249-089c8c80-d587-11ea-8ea6-1a02a6e01498.png)

Hospital tab:
![image](https://user-images.githubusercontent.com/22457859/89206281-15b97b80-d587-11ea-87b3-6055a87057b1.png)

Include Hospitals confirmation dialog:
![image](https://user-images.githubusercontent.com/22457859/89207911-7f3a8980-d589-11ea-9b4a-3ce974d35c0e.png)

Exclude Hospitals confirmation dialog: (Notice the title is plural, but the question/prompt is singular "hospital")
![image](https://user-images.githubusercontent.com/22457859/89207954-8c577880-d589-11ea-8d80-17052c13b97a.png)

Ambulance tab:
![image](https://user-images.githubusercontent.com/22457859/89206309-21a53d80-d587-11ea-9ba5-72a2c83fb541.png)

Include Ambulances confirmation dialog:
![image](https://user-images.githubusercontent.com/22457859/89207990-99746780-d589-11ea-9713-7ba5733fec77.png)

Exclude Ambulances confirmation dialog: (Notice the title is plural, but the question/prompt is singular "ambulance")
![image](https://user-images.githubusercontent.com/22457859/89208021-a2fdcf80-d589-11ea-819f-2e3eed49ed48.png)

Hamburger menu:
![image](https://user-images.githubusercontent.com/22457859/89206844-d8a1b900-d587-11ea-90d2-20eeea8cf836.png)
